qemu/tests/virtio_console.py

```diff
@@ -1792,19 +1760,26 @@ def test_unplugged_restart():
                       method=params.get('virtio_console_method', 'shell'),
                       timeout=720)
         except Exception, details:
-            raise error.TestFail("Fail to reboot VM:\n%s" % details)
+            test.fail("Fail to reboot VM:\n%s" % details)
 
         # TODO: Hotplug ports and verify that they are usable
         # VM is missing ports, which are in params.
         vm.destroy(gracefully=True)
 
-    @error.context_aware
     def test_failed_boot():
         """
         Start VM and check if it failed with the right error message.
         :param cfg: virtio_console_params - Expected error message.
         """
         exp_error_message = params['virtio_console_params']
+        version_params = params["version_params"]
+        qemu_binary = utils_misc.get_qemu_binary(params)
+        output = str(process.run(qemu_binary + " --version"))
+        re_comp = re.compile(r'\d+.\d+.\d+-\d+..+')
```

Comment:
The binary not always has a revision number (`-\d+`), for example, an upstream binary.
```
$ ./x86_64-softmmu/qemu-system-x86_64 --version
QEMU emulator version 2.9.92
```